File: 17-Jamones-Java/src/main/java/es/joseluisgs/dam/Granja.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package es.joseluisgs.dam;

/**
 *
 * @author joseluisgs
 */
public class Granja extends Thread {
    private Secadero secadero;
    private int cant, ms, prioridad;
    private String id;


    public Granja (String id, Secadero secadero, int cant, int ms, int prioridad){
        this.id = id;
        this.secadero = secadero;
        this.cant = cant;
        this.ms = ms;
        this.prioridad = prioridad;
    }

    public Granja (Secadero p){
            this.secadero = p;
    }

    public void run (){
        // Cambiamos la prioridad
        this.setPriority(this.prioridad);
        
        for (int i = 1; i < cant+1; i++){
            Jamon j = new Jamon(i,this.id);
            System.out.println("Granja "+this.id+"-> Produzco Jamón: "+i+": "+j.getId()+" de "+j.getPeso()+"KG");
            secadero.meter(j);
        }
        try {
            Thread.sleep(ms);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }    
        

    }
    
}

File: 18-Jamones-Kotlin/src/main/kotlin/jamones/models/Granja.kt
package jamones.models

import es.joseluisgs.dam.jamones.monitor.MonitorProducerConsumer

class Granja(
    private var id: String,
    private var secadero: MonitorProducerConsumer<Jamon>,
    private var cant: Int,
    private var ms: Int,
    private var prioridad: Int
) : Thread() {

    override fun run() {
        // Cambiamos la prioridad
        priority = prioridad
        for (i in 1 until cant + 1) {
            val j = Jamon(i, id)
            println("Granja " + id + "-> Produzco Jamón: " + i + ": " + j.id + " de " + j.peso + "KG")
            secadero.put(j)
        }
        try {
            sleep(ms.toLong())
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }
}